app/renderer/actions/Session.js

```diff
@@ -831,3 +835,41 @@ export function setAddVendorPrefixes (addVendorPrefixes) {
     dispatch({type: SET_ADD_VENDOR_PREFIXES, addVendorPrefixes});
   };
 }
+
+export function initFromQueryString () {
+  return async (dispatch, getState) => {
+    if (!isFirstRun) {
+      return;
+    }
+
+    isFirstRun = false;
+
+    let url = window.location.href;
+    if (url.indexOf('?') === -1) {
+      return;
+    }
+    // remove the weird anchor thing react or the router puts at the end
+    url = url.replace(/#\/$/, '');
+    const query = url.slice(url.indexOf('?') + 1);
```

Comment:
why should we do the same operation twice?

Reply:
I also assume it would be more appropriate to use [URL](https://url.spec.whatwg.org/#api) class